app/controllers/courses/age_range_controller.rb

```diff
@@ -1,11 +1,46 @@
 module Courses
   class AgeRangeController < ApplicationController
-    include EditBasicDetail
+    decorates_assigned :course
+    before_action :build_course
+
+    def edit; end
+
+    def update
+      # Age range 'other' overrides
+      course = params.dig(:course)
+      is_other = course.dig(:age_range_in_years) == "Other"
```

Comment:
You could put this logic into the `course_params` method in `age_range_controller.rb`, which is what's responsible for  what gets passed to the update action.

Then you could continue using `include EditBasicDetail`.